### PR TITLE
0.61.2 analytics

### PR DESCRIPTION
**Pull request by jamesjefferies**

### CODE REVIEW

File: app/assets/javascripts/common_chart_options.js [hunk 1]
```diff
@@ -221,32 +207,38 @@ function barColumnLine(chartData, highchartsChart, seriesData, chartConfig) {
...
-        axisTitle = 'kg/kWh';
-        pointFormat = '{point.y:.2f} kg/kWh';
-        max = 0.5;
-      } else if (y2AxisLabel == 'Solar Irradiance') {
-        axisTitle = 'Brightness of sunshine W/m2';
-        pointFormat = '{point.y:.2f} W/m2';
-      }
-      highchartsChart.addAxis({ title: { text: axisTitle }, stackLabels: { style: { fontWeight: 'bold',  color: colour }}, opposite: true, max: max});
-      highchartsChart.update({ plotOptions: { line: { tooltip: { headerFormat: '<b>{point.key}</b><br>',  pointFormat: pointFormat }}}});
+  // Handle Y2 axis
```
Comment (jamesjefferies) [inline app/assets/javascripts/common_chart_options.js]: This was being handled in two separate places already, so pulled all together, hopefully haven't missed out...

File: app/services/schedule_data_manager_service.rb [hunk 2]
```diff
@@ -13,9 +13,15 @@ def holidays
       hol_data = HolidayData.new
 
       Calendar.find(@calendar_id).outside_term_time.order(:start_date).includes(:academic_year).map do |holiday|
-        # Not really being used at the moment by the analytics code
-        academic_year = nil
-        analytics_holiday = Holiday.new(holiday.calendar_event_type.analytics_event_type.to_sym, holiday.title || 'No title', holiday.start_date, holiday.end_date, academic_year)
+        academic_year = nil # Not really being used at the moment by the analytics code
+
```
Comment (jamesjefferies) [inline app/services/schedule_data_manager_service.rb]: This is just separating out to separate lines

File: app/controllers/analysis_page_finder_controller.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+    school = School.find_by(urn: urn)
+    alert_type = AlertType.find_by(class_name: analysis_class)
+
+    if school && school.latest_analysis_pages.any?
+      analysis_page = school.latest_analysis_pages.includes(:alert).detect { |page| page.alert.alert_type_id == alert_type.id }
+    end
+
+    if analysis_page
+      redirect_to school_analysis_path(school_id: school.slug, id: analysis_page.id)
```
Comment (jamesalmond) [inline app/controllers/analysis_page_finder_controller.rb]: `school_analysis_path(schoo, analysis_page)`?

File: app/controllers/analysis_page_finder_controller.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+class AnalysisPageFinderController < ApplicationController
+  skip_before_action :authenticate_user!
+  include AnalysisPages
+
+  def show
+    urn = params[:urn]
+    analysis_class = params[:analysis_class]
+
+    school = School.find_by(urn: urn)
+    alert_type = AlertType.find_by(class_name: analysis_class)
```
Comment (jamesalmond) [inline app/controllers/analysis_page_finder_controller.rb]: `find_by!` these lookups or check the alert_type exists somewher below?

Comment (jamesjefferies) [reply app/controllers/analysis_page_finder_controller.rb]: Using find_by! now

File: app/controllers/analysis_page_finder_controller.rb (hunk 3 above)
Comment (jamesjefferies) [reply app/controllers/analysis_page_finder_controller.rb]: Yes, of course!

**Pull request closed without merging**